src/lamplib/src/genny/tasks/auto_tasks.py

```diff
@@ -488,31 +499,40 @@ class ConfigWriter:
     """
 
     @staticmethod
-    def write_config(execution: int, config: Configuration, output_file: str) -> None:
+    def write_config(
+        execution: int, config: Configuration, output_file: str, dry_run: bool = False
+    ) -> None:
         """
         :param config: The configuration to write
         :param output_file: What file to write to.
         """
-        
         success = False
         raised = None
         try:
             out_text = config.to_json()
+            if dry_run:
```

Comment:
Honestly I'd consider pulling this out into the main method, since the entire purpose of this function is to write the config, and having a parameter to not do that seems confusing and involves more threading of parameters into functions.

Reply:
I agree. I've removed dry_run from `write_config()`.